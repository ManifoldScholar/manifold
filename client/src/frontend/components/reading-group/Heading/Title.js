import React from "react";
import PropTypes from "prop-types";
import IconComposer from "global/components/utility/IconComposer";

function HeadingTitle({ groupName }) {
  return (
    <div className="group-page-heading__text-container">
      <IconComposer
        icon="readingGroup24"
        size={32}
        iconClass={"group-page-heading__icon"}
      />
      <h1 className={"heading-primary group-page-heading__text"}>
        {groupName}
      </h1>
    </div>
  );
}

HeadingTitle.displayName = "ReadingGroup.Heading.Title";

HeadingTitle.propTypes = {
  groupName: PropTypes.string.isRequired
};

export default HeadingTitle;
