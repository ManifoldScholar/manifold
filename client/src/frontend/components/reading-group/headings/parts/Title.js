import React from "react";
import PropTypes from "prop-types";
import IconComposer from "global/components/utility/IconComposer";

function HeadingTitle({ title, icon = "readingGroup24" }) {
  return (
    <div className="group-page-heading__text-container">
      <IconComposer
        icon={icon}
        size={32}
        iconClass={"group-page-heading__icon"}
      />
      <h1 className={"heading-primary group-page-heading__text"}>{title}</h1>
    </div>
  );
}

HeadingTitle.displayName = "ReadingGroup.Heading.Title";

HeadingTitle.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string
};

export default HeadingTitle;
