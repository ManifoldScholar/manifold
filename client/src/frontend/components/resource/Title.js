import React, { Component } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import FormattedDate from "global/components/FormattedDate";
import IconComputed from "global/components/icon-computed";
import Collecting from "frontend/components/collecting";

class ResourceTitle extends Component {
  static displayName = "Resource.Title";

  static propTypes = {
    resource: PropTypes.object,
    showIcon: PropTypes.bool,
    showDate: PropTypes.bool,
    t: PropTypes.func
  };

  static defaultProps = {
    showIcon: true,
    showDate: true
  };

  render() {
    const attr = this.props.resource.attributes;

    return (
      <div className="resource-title">
        {this.props.showIcon ? (
          <figure className={`resource-title__icon ${attr.kind}`}>
            <IconComputed.Resource icon={attr.kind} size={60} />
          </figure>
        ) : null}
        <div>
          <div className="resource-title__title-and-toggle">
            <h1
              className="resource-title__title"
              dangerouslySetInnerHTML={{ __html: attr.titleFormatted }}
            />
            <span className="resource-title__collecting-toggle">
              <Collecting.Toggle collectable={this.props.resource} />
            </span>
          </div>
          {this.props.showDate ? (
            <span className="resource-date">
              {this.props.t("dates.resource_added")}{" "}
              <FormattedDate format="MMMM, yyyy" date={attr.createdAt} />
            </span>
          ) : null}
        </div>
      </div>
    );
  }
}

export default withTranslation()(ResourceTitle);
