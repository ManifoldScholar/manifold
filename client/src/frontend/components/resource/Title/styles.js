import styled from "@emotion/styled";
import { respond, subtitlePrimary, fluidScale } from "theme/styles/mixins";

export const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  padding-block-end: ${fluidScale("42px", "26px")};
  text-decoration: none;
`;

export const Icon = styled.figure`
  display: none;
  padding-inline-end: 18px;

  ${respond(`display: inline-block;`, 60)}

  svg {
    fill: var(--color-base-neutral50);
  }
`;

export const TitleAndToggle = styled.div`
  display: flex;
`;

export const Title = styled.h1`
  font-family: var(--font-family-heading);
  width: 100%;
  margin: 0;
  margin-block-end: 5px;
  font-size: 26px;
  font-weight: var(--font-weight-medium);
  hyphens: none;

  ${respond(`width: auto;`, 60)}
`;

export const ToggleWrapper = styled.span`
  margin-inline-start: 12px;
  transform: translateY(5px);
`;

export const DateWrapper = styled.span`
  ${subtitlePrimary}
  display: inline-block;
  width: 100%;
  font-size: ${fluidScale("20px", "16px")};
  margin-block-start: 12px;

  .bg-neutral90 & {
    color: var(--color-base-neutral40);
  }
`;
