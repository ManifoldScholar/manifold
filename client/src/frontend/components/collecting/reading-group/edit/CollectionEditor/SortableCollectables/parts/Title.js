import React from "react";
import PropTypes from "prop-types";
import Skeleton from "frontend/components/collecting/ContentSkeleton";
import {
  getResponse,
  idInResponses
} from "frontend/components/collecting/helpers";
import * as Styled from "./styles";

function getCollectableTitle(response) {
  const { attributes, type } = response;
  if (type === "textSections")
    return `${attributes.name} <i>in ${attributes.textTitle}</i>`;
  return attributes.title;
}

function CollectableTitle({ id, responses, labelId }) {
  const isLoaded = idInResponses(id, responses);

  if (!isLoaded)
    return (
      <Skeleton
        style={{
          height: 14,
          maxWidth: "100%",
          width: 350,
          transform: "translateY(1px)"
        }}
        nested
      />
    );

  const response = getResponse(id, responses);

  return (
    <Styled.Title
      id={labelId}
      dangerouslySetInnerHTML={{ __html: getCollectableTitle(response) }}
    />
  );
}

CollectableTitle.displayName =
  "ReadingGroup.Collecting.CollectionEditor.Collectable.Title";

CollectableTitle.propTypes = {
  id: PropTypes.string.isRequired,
  responses: PropTypes.array.isRequired,
  labelId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

export default CollectableTitle;
