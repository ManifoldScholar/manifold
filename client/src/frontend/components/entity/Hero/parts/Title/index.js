import React from "react";
import PropTypes from "prop-types";
import Collecting from "frontend/components/collecting";
import * as Styled from "./styles";

export default function HeroTitle({ entity, standalone = false }) {
  const title = entity?.attributes?.titleFormatted;
  const subtitle = entity?.attributes?.subtitleFormatted;
  return (
    <Styled.Header $standalone={standalone}>
      {title && (
        <Styled.TitleWrapper>
          <Styled.Title
            dangerouslySetInnerHTML={{ __html: title }}
            $standalone={standalone}
          />
          <Styled.Toggle>
            <Collecting.Toggle collectable={entity} />
          </Styled.Toggle>
        </Styled.TitleWrapper>
      )}
      {subtitle && (
        <Styled.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
      )}
    </Styled.Header>
  );
}

HeroTitle.displayName = "Frontend.Entity.Hero.Parts.Title";

HeroTitle.propTypes = {
  entity: PropTypes.object.isRequired,
  standalone: PropTypes.bool
};
