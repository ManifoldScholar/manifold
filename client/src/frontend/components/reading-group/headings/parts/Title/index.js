import React from "react";
import PropTypes from "prop-types";
import * as Styled from "./styles";
import { useTranslation } from "react-i18next";

function HeadingTitle({
  title,
  icon = "readingGroup24",
  tag,
  adminWarning = false
}) {
  const { t } = useTranslation();

  return (
    <Styled.TextContainer>
      <Styled.Icon icon={icon} size={32} />
      <Styled.Title className="heading-primary ">
        {title}
        {adminWarning && (
          <Styled.Warning>
            {t("messages.reading_group.admin_warning")}
          </Styled.Warning>
        )}
      </Styled.Title>
      {tag && <Styled.Tag>{tag}</Styled.Tag>}
    </Styled.TextContainer>
  );
}

HeadingTitle.displayName = "ReadingGroup.Heading.Title";

HeadingTitle.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string,
  tag: PropTypes.string,
  adminWarning: PropTypes.bool
};

export default HeadingTitle;
