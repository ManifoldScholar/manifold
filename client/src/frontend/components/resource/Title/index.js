import React, { Component } from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import FormattedDate from "global/components/FormattedDate";
import IconComputed from "global/components/icon-computed";
import Collecting from "frontend/components/collecting";
import * as Styled from "./styles";

class ResourceTitle extends Component {
  static displayName = "Resource.Title";

  static propTypes = {
    resource: PropTypes.object,
    showIcon: PropTypes.bool,
    showDate: PropTypes.bool,
    t: PropTypes.func
  };

  static defaultProps = {
    showIcon: true,
    showDate: true
  };

  render() {
    const attr = this.props.resource.attributes;

    return (
      <Styled.Container>
        {this.props.showIcon ? (
          <Styled.Icon className={attr.kind}>
            <IconComputed.Resource icon={attr.kind} size={60} />
          </Styled.Icon>
        ) : null}
        <div>
          <Styled.TitleAndToggle>
            <Styled.Title
              dangerouslySetInnerHTML={{ __html: attr.titleFormatted }}
            />
            <Styled.ToggleWrapper>
              <Collecting.Toggle collectable={this.props.resource} />
            </Styled.ToggleWrapper>
          </Styled.TitleAndToggle>
          {this.props.showDate ? (
            <Styled.DateWrapper>
              {this.props.t("dates.resource_added")}{" "}
              <FormattedDate format="MMMM, yyyy" date={attr.createdAt} />
            </Styled.DateWrapper>
          ) : null}
        </div>
      </Styled.Container>
    );
  }
}

export default withTranslation()(ResourceTitle);
