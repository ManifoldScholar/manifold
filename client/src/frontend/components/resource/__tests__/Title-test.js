import Title from "../Title";

describe("frontend/components/resource/Title", () => {
  def("resource", () => factory("resource"));
  def("root", () => <Title resource={$resource} showDate showIcon />);

  it("matches the snapshot", () => {
    expect(shallow($root)).toMatchSnapshot();
  });
});
