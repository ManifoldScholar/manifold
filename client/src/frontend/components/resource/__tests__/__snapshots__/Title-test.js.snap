// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`frontend/components/resource/Title matches the snapshot 1`] = `
<div
  className="resource-title"
>
  <figure
    className="resource-icon image"
  >
    <IconComputed.Resource
      icon="image"
      size={60}
    />
  </figure>
  <div>
    <h1
      dangerouslySetInnerHTML={
        Object {
          "__html": "Image",
        }
      }
    />
    <span
      className="resource-date"
    >
      Resource added 
      <FormattedDate
        date="2017-04-24T23:25:50.161Z"
        format="MMMM, yyyy"
      />
    </span>
  </div>
</div>
`;
